include/fi_log.h

```diff
@@ -35,32 +35,107 @@
 #define FI_LOG_H
 
 #if HAVE_CONFIG_H
-#  include <config.h>
+#include <config.h>
 #endif /* HAVE_CONFIG_H */
 
-extern int fi_log_level;
+#include <fi_list.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/*
+ * Define the basic subsystems that can be used to enable fine grained logging.
+ */
+#define FABRIC (0)
+#define DOMAIN (1)
+#define CM (2)
+#define EP (3)
+#define AV (4)
+#define CQ (5)
+#define EQ (6)
+#define MR (7)
```

Comment:
Actually, I was thinking that in the final iteration I'd prefer to see these as an enumeration.